cmd/update-main.go

```diff
@@ -235,20 +256,33 @@ func getDownloadURL() (downloadURL string) {
 	return minioReleaseURL + "minio"
 }
 
-func getUpdateInfo(timeout time.Duration, mode string) (older time.Duration, downloadURL string, err error) {
-	currentReleaseTime, err := GetCurrentReleaseTime()
+// getUpdateInfo - compares latest available official release with the
+// version of current server process; returns staleness of current
+// release and download URL of latest release.
+func getUpdateInfo(mode string, timeout time.Duration) (
+	older time.Duration, downloadURL string, err error) {
+
+	u := updateCheckParams{
+		version:            Version,
+		serverBinaryPath:   os.Args[0],
+		cgroupFile:         "/proc/self/cgroup",
+		releaseChecksumURL: minioReleaseURL + "minio.shasum",
+		serverMode:         mode,
+	}
```

Comment:
You can notice that all the fields of `updateCheckParams` (except mode) are globally available and hence there is no reason for these fields to be part of this struct.

The intention of code cleanup is correct, the previous code was looking bad i.e mode was being passed all around and in the end to be used by a function somewhere down deep. This can be fixed in a better way by having `globalMinioMode` that gets set during init. Global variables are not a problem as long as no one modifies it, so having mode as globally available is fine. Also Bala is working putting all globals into a single struct so that it looks cleaner.

So you can do away with `updateCheckParams` and have mode globally available.